botorch/optim/closures.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""Model fitting closures"""
+
+from __future__ import annotations
+
+from itertools import islice
+from typing import Any, Callable, Optional, Sequence
+
+from botorch.utils.dispatcher import Dispatcher
+from gpytorch.mlls import (
+    ExactMarginalLogLikelihood,
+    MarginalLogLikelihood,
+    SumMarginalLogLikelihood,
+)
+from torch import Tensor
+from torch.utils.data import DataLoader
+
+dispatcher = Dispatcher("get_loss_closure")
+NoneType = type(None)
+TLossClosure = Callable[[], Tensor]
```

Comment:
Is this the right type if you're forwarding kwargs to the MLL?

Reply:
Not really, no. If we end up allowing `kwargs` to be passed to loss closures (debatable since `torch.jit.script` does not support this), we'll make `TLossClosure` a `typing.Protocol`.